src/io/types.ts

```diff
@@ -208,3 +208,47 @@ export interface IOHandler {
   save?: SaveHandler;
   load?: LoadHandler;
 }
+
+/**
+ * An interface for the manager of a model store.
+ *
+ * A model store is defined as a storage medium on which multiple models can
+ * be stored. Each stored model has a unique `path` as its identifier.
+ * A `ModelStoreManager` for the store allows actions including
+ *
+ * - Listing the models stored in the store.
+ * - Deleting a model from the store.
+ * - Copying a model in the store, from one path to another.
+ */
+export interface ModelStoreManager {
+  /**
+   * List all models in the model store.
+   *
+   * @returns A dictionary mapping paths of existing models to their
+   *   model artifacts info. Model artifacts info include type of the model's
+   *   topology, byte sizes of the topology, weights, etc.
+   */
+  listModels(): Promise<{[path: string]: ModelArtifactsInfo}>;
+
+  /**
+   * Delete a model specified by `path`.
+   *
+   * @param path
+   * @returns ModelArtifactsInfo of the deleted model (if and only if deletion
```

Comment:
Why does deleting return the ModelArtifactsInfo?    Is this a common pattern for the deletion of assets in JS?  Are we creating a difficult-to-fix problem if some storage medium can sense presence & delete much faster than access the ModelArtifactsInfo?